From a55161c1649528b7109e7168237a6657ecf9052c Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 1 Aug 2017 14:27:34 +0800
Subject: [PATCH 2267/5242] arm64: dts: freescale: imx8qm: add reserve memory
 for rpmsg

commit  eff4b2609392a10c7ba7ee787fec51da5bcf7458 from
https://source.codeaurora.org/external/imx/linux-imx.git

RPMSG needs to share memory with M4, and SCD needs to set
this shared memory property accordingly, so the memory
region needs to be fixed.

Here make CMA range same as its size, so that SCD only sets
this region as share property. And also reserve another
4MB for RPMSG.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi  |    7 ++++++-
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |    7 ++++++-
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index 4a0fe49..12b3c80 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -73,9 +73,14 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			size = <0 0x28000000>;
-			alloc-ranges = <0 0x80000000 0 0x80000000>;
+			alloc-ranges = <0 0x90000000 0 0x28000000>;
 			linux,cma-default;
 		};
+
+		rpmsg_reserved: rpmsg@0xb8000000 {
+			no-map;
+			reg = <0 0xb8000000 0 0x400000>;
+		};
 	};
 
 	gic: interrupt-controller@51a00000 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 09042fd..2f12c33 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -68,9 +68,14 @@
 			compatible = "shared-dma-pool";
 			reusable;
 			size = <0 0x28000000>;
-			alloc-ranges = <0 0x80000000 0 0x80000000>;
+			alloc-ranges = <0 0x90000000 0 0x28000000>;
 			linux,cma-default;
 		};
+
+		rpmsg_reserved: rpmsg@0xb8000000 {
+			no-map;
+			reg = <0 0xb8000000 0 0x400000>;
+		};
 	};
 
 	gic: interrupt-controller@51a00000 {
-- 
1.7.9.5

